src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -105,11 +110,69 @@ Transaction *TimestampOrderingTransactionManager::BeginTransaction() {
 }
 
 Transaction *TimestampOrderingTransactionManager::BeginReadonlyTransaction() {
+  PL_ASSERT(is_from_transaction_pool_ == false);
+  
+  auto &epoch_manager = EpochManagerFactory::GetInstance();
+
   txn_id_t txn_id = READONLY_TXN_ID;
+  cid_t begin_cid = epoch_manager.GetReadOnlyTxnCid();
+  
+  Transaction *txn = new Transaction(txn_id);
```

Comment:
Is there a dependency on the networking threads being separated from worker threads?